flecsi/execution/hpx/execution_policy.h

```diff
@@ -219,9 +242,29 @@ struct FLECSI_EXPORT hpx_execution_policy_t {
       return future;
     };
 
-    // for unwrapping o returned future
-    hpx_future_u<RETURN> future = hpx::async(std::move(func));
-    future.wait();
+    // force unwrapping of returned future
+    hpx_future_u<RETURN> future = hpx::dataflow(
```

Comment:
@hkaiser 

How asynchronous are hpx::dataflow and hpx::future.then? Do they just submit a "task" and its continuation and return immediately?

Reply:
> How asynchronous are hpx::dataflow and hpx::future.then? Do they just submit a "task" and its continuation and return immediately?

Yes, that's exactly what they do. They will schedule a new task once the futures they depend on have become ready.